### PR TITLE
docs: [M3-9010] - `useEffect` Best Practices

### PR DESCRIPTION
**Pull request by jaalah-akamai**
## Description 📝
Updating our docs regarding UseEffect and how they should only be used for true side effects.

## Target release date 🗓️
N/A

### CODE REVIEW

File: docs/development-guide/13-coding-standards.md [hunk 1]
```diff
@@ -14,7 +14,19 @@ We use [ESLint](https://eslint.org/) to enforce coding and formatting standards.
 If you are using VSCode it is **highly** recommended to use the [ESlint extension](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint). The [Prettier extension](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode) is also recommended, as it can be configured to format your code on save.
 
 ## React
+### useEffect()
+UseEffect should only be used for handling true side effects - specifically API calls, subscriptions, and DOM mutations that must occur outside React's render cycle. While you may encounter UseEffect throughout our existing codebase, we're actively working to remove these instances. Don't use existing code as precedent for implementing new UseEffects. All state updates and data transformations should be handled through event handlers and direct state management.
```
Comment (dwiley-akamai) [inline docs/development-guide/13-coding-standards.md:18]:
```suggestion
`useEffect()` should only be used for handling true side effects - specifically API calls, subscriptions, and DOM mutations that must occur outside React's render cycle. While you may encounter instances where `useEffect()` is used differently throughout our existing codebase, we're actively working to remove those instances. Existing code that does not adhere to the hook's proper use should not be used as precedent for implementing new `useEffect()` instances. All state updates and data transformations should be handled through event handlers and direct state management.
```

**Review by dwiley-akamai — COMMENTED** [review]
The ticket calls for before/after code examples, are those coming in a follow-up PR?

**Review by bnussman-akamai — APPROVED** [review]
Thanks for adding this 🙏

**Comment by jaalah-akamai** [comment]
> The ticket calls for before/after code examples, are those coming in a follow-up PR?

I thought about that, but the React docs provide some pretty clear examples. I may find some examples of instances involving React Hook Form where devs thought they may have needed a useEffect when they didn't

**Review by dwiley-akamai — APPROVED** [review]

**Pull request merged**